saltproc/openmc_depcode.py

```diff
@@ -189,24 +434,32 @@ def write_runtime_input(self, reactor, depletion_step, restart):
         """
 
         if depletion_step == 0 and not restart:
+            geo_file = self.geo_file_paths.pop(0)
+            openmc.reset_auto_ids()
             materials = openmc.Materials.from_xml(
                 self.template_input_file_path['materials'])
             geometry = openmc.Geometry.from_xml(
-                self.geo_file_paths[0], materials=materials)
+                geo_file, materials=materials)
             settings = openmc.Settings.from_xml(
                 self.template_input_file_path['settings'])
             self.npop = settings.particles
             self.inactive_cycles = settings.inactive
             self.active_cycles = settings.batches - self.inactive_cycles
 
         else:
+            openmc.reset_auto_ids()
             materials = openmc.Materials.from_xml(self.runtime_matfile)
             geometry = openmc.Geometry.from_xml(
                 self.runtime_inputfile['geometry'], materials=materials)
             settings = openmc.Settings.from_xml(
                 self.runtime_inputfile['settings'])
 
-        materials.export_to_xml(self.runtime_matfile)
+        diluted_model = openmc.Model(materials=materials, geometry=geometry, settings=settings)
+        reactions, diluted_materials = MicroXS._add_dilute_nuclides(self.chain_file_path,
+                                                                   diluted_model,
+                                                                   1e3)
```

Comment:
```
    @classmethod
    def _add_dilute_nuclides(cls, chain_file, model, dilute_initial):
        """
        Add nuclides not present in burnable materials that have neutron data
        and are present in the depletion chain to those materials. This allows
        us to tally those specific nuclides for reactions to create one-group
        cross sections.

        Parameters
        ----------
        chain_file : str
            Path to the depletion chain XML file that will be used in depletion
            simulation. Used to determine cross sections for materials not
            present in the inital composition.
        model : openmc.Model
            Model object
        dilute_initial : float
            Initial atom density [atoms/cm^3] to add for nuclides that
            are zero in initial condition to ensure they exist in the cross
            section data. Only done for nuclides with reaction rates.

        Returns
        -------
        reactions : list of str
            List of reaction names
        diluted_materials : openmc.Materials
            :class:`openmc.Materials` object with nuclides added to burnable
            materials.
        """
```

Reply:
I think it might be helpful to include a variable `inital_atom_dens = 1e3` or something similar, but if that won't need to be changed later then feel free to leave it as is.

Reply:
eh, it's not really necessary in my opinion..